### PR TITLE
feat: service type summary

### PR DESCRIPTION
**Pull request by amosmachora**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).
- [ ] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
In this PR I have added an additional filter by service type to the payment history component. This allows for users to see their bills breakdown by service type. For more context please see [here.](https://github.com/palladiumkenya/openmrs-module-kenyaEMRCashier/pull/39#issuecomment-2376336573) You will understand the necessity of this feature and how it came about.

## Screenshots
https://github.com/user-attachments/assets/3c653bfe-8013-4cb8-babb-162bd7a67f36

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

**Comment by ojwanganto** [comment]
Thanks, @amosmachora. Over to the experienced guys

File: packages/esm-billing-app/src/billable-services/payment-history/payment-history-table.component.tsx [hunk 1]
```diff
@@ -1,26 +1,20 @@
...
+import styles from './payment-history.scss';
 
 export const PaymentHistoryTable = ({
   tableData,
   paidBillsResponse,
+  renderedRows,
 }: {
   tableData: any;
   paidBillsResponse: ReturnType<typeof useBills>;
+  renderedRows: MappedBill[];
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/payment-history/payment-history-table.component.tsx:17]:
```suggestion
  renderedRows: Array<MappedBill>;
```
For constistency lets have array declaration following this syntax

```ts
Array<T>
```

File: packages/esm-billing-app/src/billable-services/payment-history/payment-history-viewer.component.tsx [hunk 2]
```diff
@@ -48,6 +49,7 @@ const PaymentHistoryViewer = () => {
 
   const [selectedPaymentTypeCheckBoxes, setSelectedPaymentTypeCheckBoxes] = useState<string[]>([]);
   const [selectedCashierCheckboxes, setSelectedCashierCheckboxes] = useState<string[]>([]);
+  const [selectedServiceTypeCheckboxes, setSelectedServiceTypeCheckboxes] = useState<string[]>([]);
```
Comment (donaldkibet) [inline packages/esm-billing-app/src/billable-services/payment-history/payment-history-viewer.component.tsx:52]: I would recommend we use `useReducer` hook here to manager filter states. IMO its simpler and cleaner

**Review by donaldkibet — APPROVED** [review]
Thanks left some suggestions

File: packages/esm-billing-app/src/billable-services/payment-history/payment-history-table.component.tsx (hunk 1 above)
Comment (amosmachora) [reply packages/esm-billing-app/src/billable-services/payment-history/payment-history-table.component.tsx:17]: Got it. From now onwards I will use the syntax.

File: packages/esm-billing-app/src/billable-services/payment-history/payment-history-viewer.component.tsx (hunk 2 above)
Comment (amosmachora) [reply packages/esm-billing-app/src/billable-services/payment-history/payment-history-viewer.component.tsx:52]: I will open a follow up PR on this. I have not used useReducer so I will have to learn then I will open a follow up PR.

**Pull request merged**